src/Features/Core/Portable/ExtractMethod/SelectionResult.cs

```diff
@@ -15,213 +13,203 @@
 
 namespace Microsoft.CodeAnalysis.ExtractMethod;
 
-/// <summary>
-/// clean up this code when we do selection validator work.
-/// </summary>
-internal abstract class SelectionResult<TStatementSyntax>(
-    TextSpan originalSpan,
-    TextSpan finalSpan,
-    bool selectionInExpression,
-    SemanticDocument document,
-    SyntaxAnnotation firstTokenAnnotation,
-    SyntaxAnnotation lastTokenAnnotation,
-    bool selectionChanged)
-    where TStatementSyntax : SyntaxNode
+internal enum SelectionType
 {
-    private bool? _createAsyncMethod;
+    Expression,
+    SingleStatement,
+    MultipleStatements,
+}
 
-    public TextSpan OriginalSpan { get; } = originalSpan;
-    public TextSpan FinalSpan { get; } = finalSpan;
-    public bool SelectionInExpression { get; } = selectionInExpression;
```

Comment:
removed.  'SelectionType' is exposed instead (and 3 properties based off of that).